[EMGR-438] Define V1 and V2 Metrics and their Differences (#516)
hazelcast/management-center-docs: docs/modules/integrate/pages/prometheus-metrics.adoc

{description}

> By default, Management Center exposes both the original metrics and the updated versions. Existing integrations will therefore continue to work, but you should consider the original metrics deprecated and plan to migrate to the updated versions. You can optionally xref:config[change which set of metrics is exposed].

Reviewer: Did we change policy on this?

Author: V2 is still in BETA and undergoing breaking changes, e.g. in 5.8 some metric names were changed in V2. So, the original (V1) cannot be deprecated yet as there's nothing to migrate to that's stable -- V2.

When V2 has a release version we can deprecate the V1 metrics officially, perhaps using some doc string in the metrics themselves so that downstream users can some indication, e.g.

```
# HELP some_counter (Deprecated since 5.9.0) this counts things
```

* Version 1 (V1) Metrics. Take the metrics exposed by {page-latest-supported-hazelcast}@hazelcast::list-of-metrics.adoc[Hazelcast's JMX metrics system] and applies the following transformation to each of the metric names:
  * Substitutes `.` for `_`
  * Adds the prefix `hz_`
* Version 2 (V2) Metrics. Take the metrics exposed by {page-latest-supported-hazelcast}@hazelcast::list-of-metrics.adoc[Hazelcast's JMX metrics system] but applies transformation to their names so that they conform with link:https://prometheus.io/docs/practices/naming/[Prometheus naming conventions]:
  * Use `snake_case`
  * Add `hazelcast_` prefix
  * Update metric names where appropriate
  * Combine related metrics and add labels
  * Convert to recommended base units:
  ** Time in seconds
  ** Memory in bytes
  ** Percentages as ratios (for example, 50% converts to 0.5)
  * Append units to metric names
  * Remove unit labels
  * Append `_total` to accumulated metrics
  * Rename and remove `unit` label, add `_total` suffix:

> We currently publish by default V1 and V2 metrics; however, the V1 metrics are marked for deprecation and will be removed in an upcoming release of Management Centre.

Reviewer 2: ```suggestion
We currently publish by default V1 and V2 metrics; however, the V1 metrics are marked for deprecation and will be removed in an upcoming release of Management Center.
```

Author: Thanks @amandalindsay -- I actually changed this in a commit just after the one you reviewed.

> We currently publish by default V1 and V2 metrics.

Reviewer: ```suggestion
```

Already stated

```

* Rename and append unit, remove `unit` label, convert value (1.7x10^12^ milliseconds to 1.7x10^9^ seconds):
+
```
# V1
hz_map_creationTime{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",unit="MS",} 1.737715861118E12 1737715903399
# V2
hazelcast_map_creation_timestamp_seconds{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",} 1.737715861118E9 1737715903399
```

* Rename and append unit, remove `unit` label, add `remove` label:
+
```
# V1
hz_map_totalMaxRemoveLatency{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",unit="MS",} 0.0 1737715903399
# V2
hazelcast_map_latency_max_seconds{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",operation="remove",} 0.0 1737715903399
```

* Rename, append unit suffix and `_total` suffix, remove `unit` label, add `remove` label, convert value from milliseconds to seconds:
+
```
# V1
hz_map_totalPutLatency{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",unit="MS",} 2019.0 1743601193973
# V2
hazelcast_map_latency_seconds_total{name="map-1",mc_member="127.0.0.1:5701",mc_cluster="Cluster-1",operation="put",} 2.019 1743601193973
```

[[config]]

> == Configuring the Metrics Version

Reviewer: ```suggestion
== Configuring the metrics version
```
We're moving to title case across the docs in line with the Google style guide

> You can tell Management Center which metrics version to export by defining the system property `hazelcast.mc.prometheusExporter.printers` as follows:

Reviewer: ```suggestion
To change which set of metrics is exposed to Prometheus, update the `hazelcast.mc.prometheusExporter.printers` system property:
```
Can only really tell because they're side by side, but I think the original text is subtly more appropriate here. The new text implies this is an action you have to take - to tell MC one way or the other. "change" implies changing the default behavior.

> You can tell Management Centre which metrics version to export by defining the system property `hazelcast.mc.prometheusExporter.printers` as follows:

Reviewer 2: ```suggestion
You can tell Management Center which metrics version to export by defining the system property `hazelcast.mc.prometheusExporter.printers` as follows:
```

Author: https://github.com/hazelcast/management-center-docs/pull/516/commits/098cb3edc54c9265aef5fe04a8ac1192381d472f

* *Version 1 (V1) Metrics.* Take the metrics exposed by xref:{page-latest-supported-hazelcast}@hazelcast::list-of-metrics.adoc[Hazelcast's JMX metrics system] and applies the following transformation to each of the metric names:
** Substitutes `.` for `_`
** Adds the prefix `hz_`

> * *Version 2 (V2) Metrics.* _Currently in BETA_. Take the metrics exposed by xref:{page-latest-supported-hazelcast}@hazelcast::list-of-metrics.adoc[Hazelcast's JMX metrics system] but applies transformation to their names so that they conform with link:https://prometheus.io/docs/practices/naming/[Prometheus naming conventions]:

Author: As discussed -- only V2 is BETA so page level beta would be incorrect.
